Makefile

```diff
@@ -1,49 +1,60 @@
 ENV_DIR=.env
+PYTHON=python3
 
 ifeq ($(OS),Windows_NT)
 	IN_ENV=. $(ENV_DIR)/Scripts/activate &&
 else
 	IN_ENV=. $(ENV_DIR)/bin/activate &&
 endif
 
+# Some distros need to use pip3 as the entrypoint
+ifneq (, $(shell which pip3))
+PIP_CMD=pip3
+else ifneq (, $(shell which pip))
+PIP_CMD=pip
+else
+$(error "Python's pip not found on $(PATH)")
+endif
+
 all: test lint docs artifacts
 
 env: $(ENV_DIR)
 
 test: build
 	$(IN_ENV) tox
 
-artifacts: build_reqs sdist wheel
+artifacts: build-reqs sdist wheel
 
 $(ENV_DIR):
-	virtualenv -p python $(ENV_DIR)
+	virtualenv -p $(PYTHON) $(ENV_DIR)
 
-build_reqs: env
-	$(IN_ENV) pip install sphinx pep8 coverage nose wheel twine tox
+build-reqs: env
+	$(IN_ENV) $(PIP_CMD) install build sphinx wheel twine setuptools_scm
 
-build: build_reqs
-	$(IN_ENV) pip install --editable .
+build: build-reqs
+	$(IN_ENV) $(PIP_CMD) install --editable .[dev]
 
-sdist: build_reqs
-	$(IN_ENV) python setup.py sdist
+sdist: build-reqs
+	$(IN_ENV) $(PYTHON) -m build --no-isolation --sdist
 
-wheel: build_reqs
-	$(IN_ENV) python setup.py bdist_wheel
+wheel: build-reqs
+	$(IN_ENV) $(PYTHON) -m build --no-isolation --wheel
 
-lint: pep8
+format-code:
```

Comment:
Love black formatting!

Reply:
I agree with black "most of the time" - and that's good enough for me. Yay for not having to think about it at all!